exec/java-exec/src/test/java/org/apache/drill/test/rowSet/file/JsonFileBuilder.java

```diff
@@ -0,0 +1,159 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.drill.test.rowSet.file;
+
+import com.google.common.base.Preconditions;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.Lists;
+import com.google.common.collect.Maps;
+import org.apache.drill.exec.record.MaterializedField;
+import org.apache.drill.exec.vector.accessor.ColumnAccessor;
+import org.apache.drill.exec.vector.accessor.ColumnReader;
+import org.apache.drill.test.rowSet.RowSet;
+
+import java.io.BufferedOutputStream;
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+
+public class JsonFileBuilder
+{
+  public static final String DEFAULT_DOUBLE_FORMATTER = "%f";
+  public static final String DEFAULT_INTEGER_FORMATTER = "%d";
+  public static final String DEFAULT_LONG_FORMATTER = "%d";
+  public static final String DEFAULT_STRING_FORMATTER = "\"%s\"";
+  public static final String DEFAULT_DECIMAL_FORMATTER = "%s";
+  public static final String DEFAULT_PERIOD_FORMATTER = "%s";
+
+  public static final Map<String, String> DEFAULT_FORMATTERS = new ImmutableMap.Builder()
+    .put(ColumnAccessor.ValueType.DOUBLE, DEFAULT_DOUBLE_FORMATTER)
+    .put(ColumnAccessor.ValueType.INTEGER, DEFAULT_INTEGER_FORMATTER)
+    .put(ColumnAccessor.ValueType.LONG, DEFAULT_LONG_FORMATTER)
+    .put(ColumnAccessor.ValueType.STRING, DEFAULT_STRING_FORMATTER)
+    .put(ColumnAccessor.ValueType.DECIMAL, DEFAULT_DECIMAL_FORMATTER)
+    .put(ColumnAccessor.ValueType.PERIOD, DEFAULT_PERIOD_FORMATTER)
+    .build();
+
+  private final RowSet rowSet;
+  private final Map<String, String> customFormatters = Maps.newHashMap();
+
+  public JsonFileBuilder(RowSet rowSet) {
+    this.rowSet = Preconditions.checkNotNull(rowSet);
+    Preconditions.checkArgument(rowSet.rowCount() > 0, "The given rowset is empty.");
+  }
+
+  public JsonFileBuilder setCustomFormatter(final String columnName, final String columnFormatter) {
+    Preconditions.checkNotNull(columnName);
+    Preconditions.checkNotNull(columnFormatter);
+
+    Iterator<MaterializedField> fields = rowSet
+      .schema()
+      .batch()
+      .iterator();
+
+    boolean hasColumn = false;
+
+    while (!hasColumn && fields.hasNext()) {
+      hasColumn = fields.next()
+        .getName()
+        .equals(columnName);
+    }
+
+    final String message = String.format("(%s) is not a valid column", columnName);
+    Preconditions.checkArgument(hasColumn, message);
+
+    customFormatters.put(columnName, columnFormatter);
+
+    return this;
+  }
+
+  public void build(File tableFile) throws IOException {
```

Comment:
Great! This does not yet handle nested tuples or arrays; in part because the row set work for that is still sitting in PR #914. You can update this to be aware of maps and map arrays once that PR is committed.

Reply:
Sounds Good